src/main/scala/com/github/codelionx/distod/actors/ResultCollector.scala

```diff
@@ -0,0 +1,99 @@
+package com.github.codelionx.distod.actors
+
+import java.io.{BufferedWriter, FileWriter}
+
+import akka.actor.typed.{Behavior, PostStop}
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import akka.actor.ActorPath
+import akka.actor.typed.receptionist.{Receptionist, ServiceKey}
+import com.github.codelionx.distod.protocols.ResultCollectionProtocol.{AckBatch, DependencyBatch, ResultCommand}
+import com.github.codelionx.distod.Settings
+import com.github.codelionx.distod.types.{OrderDependency, PendingJobMap}
+
+
+object ResultCollector {
+
+  val CollectorServiceKey: ServiceKey[ResultCommand] = ServiceKey("resultCollector")
+
+  val name = "result-collector"
+
+  def apply(): Behavior[ResultCommand] = Behaviors.setup { context =>
+    new ResultCollector(context).start()
+  }
+}
+
+
+class ResultCollector(context: ActorContext[ResultCommand]) {
+
+  private val settings: Settings = Settings(context.system)
+
+  protected def createWriter(append: Boolean): BufferedWriter =
+    new BufferedWriter(new FileWriter(settings.outputFilePath, append))
+
+  private def recreateWriter(currentWriter: Option[BufferedWriter], append: Boolean): BufferedWriter = {
+    cleanup(currentWriter)
+    createWriter(append)
+  }
+
+  private def cleanup(writer: Option[BufferedWriter]): Unit = writer match {
+    case Some(writer) =>
+      try {
+        writer.flush()
+        writer.close()
+      } catch {
+        case e: java.io.IOException =>
+          context.log.error("Could not flush and close file writer, because {}", e)
+      }
+    case None => // do nothing
+  }
+
+  @throws[java.io.IOException]
+  private def writeBatch(deps: Seq[OrderDependency], writer: BufferedWriter): Unit = {
+    for (dep <- deps) {
+      writer.write(dep.toString)
+      writer.write("\n")
+      if (settings.outputToConsole)
+        context.log.info("Found OD: {}", dep)
+    }
+    writer.flush()
+  }
+
+  def start(): Behavior[ResultCommand] = {
+    // register at the registry
+    context.system.receptionist ! Receptionist.Register(ResultCollector.CollectorServiceKey, context.self)
+
+    behavior(recreateWriter(None, append = false), PendingJobMap.empty)
+  }
+
+  private def behavior(
+      writer: BufferedWriter, seenBatches: PendingJobMap[ActorPath, Int]
+  ): Behavior[ResultCommand] = Behaviors
+    .receiveMessage[ResultCommand] { case DependencyBatch(id, deps, ackTo) =>
+      val wasSeen = seenBatches.get(ackTo.path).fold(false)(_.contains(id))
+      if (wasSeen) {
+        context.log.warn("Ignoring duplicated batch {} from {}", id, ackTo)
+        ackTo ! AckBatch(id)
+        Behaviors.same[ResultCommand]
+      } else {
+        def handleBatchWith(w: BufferedWriter): Behavior[ResultCommand] = {
+          writeBatch(deps, w)
+          ackTo ! AckBatch(id)
+          behavior(w, seenBatches + (ackTo.path -> id))
+        }
+
+        try {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using util.Try instead of a try-catch block](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4537008)